alacritty/src/daemon.rs

```diff
@@ -1,34 +1,53 @@
 use std::ffi::OsStr;
+use std::fmt::Debug;
 use std::io;
-use std::process::{Command, Stdio};
-
 #[cfg(not(windows))]
 use std::os::unix::process::CommandExt;
-
 #[cfg(windows)]
 use std::os::windows::process::CommandExt;
+use std::process::{Command, Stdio};
+
+use log::{debug, warn};
+
 #[cfg(windows)]
 use winapi::um::winbase::{CREATE_NEW_PROCESS_GROUP, CREATE_NO_WINDOW};
 
-/// Threading utilities.
-pub mod thread {
-    /// Like `thread::spawn`, but with a `name` argument.
-    pub fn spawn_named<F, T, S>(name: S, f: F) -> std::thread::JoinHandle<T>
-    where
-        F: FnOnce() -> T + Send + 'static,
-        T: Send + 'static,
-        S: Into<String>,
-    {
-        std::thread::Builder::new().name(name.into()).spawn(f).expect("thread spawn works")
-    }
+/// Start the daemon and log error on failure.
+pub fn start_daemon<I, S>(program: &str, args: I)
+where
+    I: IntoIterator<Item = S> + Debug + Copy,
+    S: AsRef<OsStr>,
+{
+    match start_daemon_impl(program, args) {
+        Ok(_) => debug!("Launched {} with args {:?}", program, args),
+        Err(_) => warn!("Unable to launch {} with args {:?}", program, args),
+    };
+}
 
-    pub use std::thread::*;
+#[cfg(windows)]
+fn start_daemon_impl<I, S>(program: &str, args: I) -> io::Result<()>
+where
+    I: IntoIterator<Item = S> + Copy,
+    S: AsRef<OsStr>,
+{
+    // Setting all the I/O handles to null and setting the
+    // CREATE_NEW_PROCESS_GROUP and CREATE_NO_WINDOW has the effect
+    // that console applications will run without opening a new
+    // console window.
+    Command::new(program)
+        .args(args)
+        .stdin(Stdio::null())
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .creation_flags(CREATE_NEW_PROCESS_GROUP | CREATE_NO_WINDOW)
+        .spawn()
+        .map(|_| ())
 }
 
 #[cfg(not(windows))]
-pub fn start_daemon<I, S>(program: &str, args: I) -> io::Result<()>
+fn start_daemon_impl<I, S>(program: &str, args: I) -> io::Result<()>
 where
-    I: IntoIterator<Item = S>,
+    I: IntoIterator<Item = S> + Copy,
```

Comment:
Why did you add the `Copy` constraint?

Reply:
Because you should copy a reference? Otherwise it's not working, since you should print it as well and to pass it somewhere else?